src/instructlab/model/evaluate.py

```diff
@@ -542,122 +467,204 @@ def evaluate(
-            s = score["score"]
-            b_s = base_score["score"]
-            d = round(s - b_s, 2)
-            if s > b_s:
-                improvements.append((task, d))
-            elif b_s > s:
-                regressions.append((task, d))
+            print("# KNOWLEDGE EVALUATION REPORT")
+            display_model(model)
+            print("\n### AVERAGE:")
+            print(f"{round(overall_score, 2)} (across {len(individual_scores)})\n")
+
+            print("### SCORES:")
+            for task, score in individual_scores.items():
+                s = round(score["score"], 2)
+                print(f"{task} - {s}")
+
+        elif benchmark == Benchmark.MMLU_BRANCH:
+            # Third Party
+            from instructlab.eval.mmlu import MMLUBranchEvaluator
+
+            evaluators = [
+                evaluator,
+                MMLUBranchEvaluator(
+                    base_model,
+                    tasks_dir,
+                    ["mmlu_pr"],
+                    few_shots=few_shots,
+                    batch_size=batch_size,
+                ),
+            ]
+            m_paths = [model, base_model]
+            overall_scores = []
+            individual_scores_list = []
+            for evaluator in evaluators:
+                overall_score, individual_scores = evaluator.run()
+                overall_scores.append(overall_score)
+                individual_scores_list.append(individual_scores)
+
+            overall_score = overall_scores[0]
+            base_overall_score = overall_scores[1]
+            individual_scores = individual_scores_list[0]
+            base_individual_scores = individual_scores_list[1]
+
+            print("# KNOWLEDGE EVALUATION REPORT\n")
+            display_models(model, base_model)
+
+            print("\n### AVERAGE:")
+            delta = round(overall_score - base_overall_score, 2)
+            if delta >= 0:
+                delta_display = f"+{delta}"
             else:
-                no_changes.append(task)
+                delta_display = delta
+
+            print(f"{delta_display} (across {len(individual_scores)})")
+
+            improvements, regressions, no_changes = [], [], []
+            for task, score in individual_scores.items():
+                base_score = base_individual_scores[task]
+                s = score["score"]
+                b_s = base_score["score"]
+                d = round(s - b_s, 2)
+                if s > b_s:
+                    improvements.append((task, d))
+                elif b_s > s:
+                    regressions.append((task, d))
+                else:
+                    no_changes.append(task)
 
-        # display summary of evaluation before exiting
-        display_branch_eval_summary(improvements, regressions, no_changes)
+            # display summary of evaluation before exiting
+            display_branch_eval_summary(improvements, regressions, no_changes)
+    except EvalError as ee:
```

Comment:
I see, so we are catching any and all errors here? That seems like a good first step

Reply:
It's catching everything known from eval.